stub/src/main/java/io/grpc/stub/ServerCalls.java

```diff
@@ -66,9 +133,6 @@ public void onPayload(ReqT request) {
               // We delay calling method.invoke() until onHalfClose(), because application may call
               // close(OK) inside invoke(), while close(OK) is not allowed before onHalfClose().
               this.request = request;
-
-              // Request delivery of the next inbound message.
-              call.request(1);
```

Comment:
So this will make it so that we no longer get into the else block. Can you change the first `call.request(1)` to request a larger number (i.e., 2)?